### ISSUE TITLE
SANA support

### ISSUE THREAD

**Issue by recoilme** [kohya-ss/sd-scripts#1807]
May you please add minimal SANA support?
https://github.com/NVlabs/Sana

SANA train implementation very not optimal(
https://github.com/NVlabs/Sana/issues/49

GPU intensive, has no cache/ Multi Aspect Ratio / Adafactor support(

We love you, mr Kohya!

**Comment by CorradoF** [comment]
> Sana is very interesting. However, I am concerned that the weight license is CC BY-NC-SA 4.0. If this applies not only to the model but also to the generated images, the use cases for the model will be limited.

What's the difference to flux.1dev? Isn't it research only too?

**Comment by CorradoF** [comment]
> > What's the difference to flux.1dev? Isn't it research only too?
> 
> From my understanding, the output of FLUX.1 dev is not covered by the license of FLUX.1 dev.
> 
> https://github.com/black-forest-labs/flux/blob/main/model_licenses/LICENSE-FLUX1-dev
> 
> > a. “Derivative”  means any (i) modified version of the FLUX.1 [dev] Model (including but not limited to any customized or fine-tuned version thereof), (ii) work based on the FLUX.1 [dev] Model, or (iii) any other derivative work thereof. For the avoidance of doubt, Outputs are not considered Derivatives under this License.

Interesting thank you, I always heard differently on reddit, my bad for not reading it myself.

If Nvidia doesn't change the license eventually there will be a pony sana. Having things already ready for that could be an advantage. Thank you anyway for the great work you did so far

**Comment by recoilme** [comment]
> Sana is very interesting. However, I am concerned that the weight license is CC BY-NC-SA 4.0.

Looks like training code under Apache now, not sure
https://github.com/NVlabs/Sana/commit/335d4452126952b90376a6f7ccd0ce0490a16fa9

**Comment by Bocchi-Chan2023** [comment]
https://github.com/NVlabs/Sana/issues/54
code only, unfortunately.

**Comment by recoilme** [comment]
@Muinez add very good code for MAR bucketing and simplified local train:
https://github.com/Muinez/Sana

i add code for model load/save and fix some small bugs
https://github.com/recoilme/Sana

And example how to use it and train small model from zero in bf16
https://github.com/recoilme/Sana/blob/main/TRAIN.md

**Comment by Bocchi-Chan2023** [comment]
> @Muinez add very good code for MAR bucketing and simplified local train: https://github.com/Muinez/Sana
> 
> i add code for model load/save and fix some small bugs https://github.com/recoilme/Sana
> 
> And example how to use it and train small model from zero in bf16 https://github.com/recoilme/Sana/blob/main/TRAIN.md

Is it possible to improve the way the dataset cache is generated? When I tried the official code, it used all the system RAM and froze my PC.

**Comment by recoilme** [comment]
> Is it possible to improve the way the dataset cache is generated?
official code dont have cache

**Comment by Bocchi-Chan2023** [comment]
> > Is it possible to improve the way the dataset cache is generated?
> 
> official code dont have cache

Oh, maybe I'm misunderstanding something.
It started using several times more memory than Sigma fine tuning, so I feel something is wrong

**Comment by recoilme** [comment]
current status
0.6b on A40(48Gpu) with 256 batch 512-1024 resolution
1.6b with 16 batch 512-2048 res

looks like success train from zero in bf16
https://wandb.ai/recoilme/potato/runs/8?nw=nwuserrecoilme

its not official training code.

**Comment by Manni1000** [comment]
but i have seen a guy that said it runns with 24gb vram if you use vae cashing is this true or false?

**Comment by recoilme** [comment]
> but i have seen a guy that said it runns with 24gb vram if you use vae cashing is this true or false?

true, i train batch 256 0.6b and batch 24 1.6b on 48 gpu with vae cache

**Comment by sdbds** [comment]
Officially released the diffusers code, should we start following up?
https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/README_sana.md

**Comment by cvtower** [comment]
+1
official dreambooth lora training might not be so good as sd-scripts repo, according my tests.

**Comment by recoilme** [comment]
We find Sana not trainable to 2024 quality standards and develop new model

https://github.com/recoilme/waifu

So we don't need sana more, but very introsting in waifu support)

**Comment by Bocchi-Chan2023** [comment]
> We find Sana not trainable to 2024 quality standards and develop new model
> 
> https://github.com/recoilme/waifu
> 
> So we don't need sana more, but very introsting in waifu support)

Don't advertise your own unrelated projects in the issue.

**Comment by recoilme** [comment]
its related project, its based on sana train, compatible with sana (just change config) and we implemented :
cache/ Multi Aspect Ratio / Adafactor support
missing in sana train

it may be reused

**Comment by Pevernow** [comment]
> > What's the difference to flux.1dev? Isn't it research only too?
> 
> From my understanding, the output of FLUX.1 dev is not covered by the license of FLUX.1 dev.
> 
> https://github.com/black-forest-labs/flux/blob/main/model_licenses/LICENSE-FLUX1-dev
> 
> > a. “Derivative”  means any (i) modified version of the FLUX.1 [dev] Model (including but not limited to any customized or fine-tuned version thereof), (ii) work based on the FLUX.1 [dev] Model, or (iii) any other derivative work thereof. For the avoidance of doubt, Outputs are not considered Derivatives under this License.

At least this license does not conflict with this project.
Onetrainer already supports it, why don't we?

**Comment by Pevernow** [comment]
> Sana is very interesting. However, I am concerned that the weight license is CC BY-NC-SA 4.0. If this applies not only to the model but also to the generated images, the use cases for the model will be limited.
> 
> It seems that a pull request for optimization is being developed in the Sana repository, so I will look forward to that first.

At least this license does not conflict with this project.
Onetrainer already supports it, why don't we?

**Comment by Pevernow** [comment]
> Sana is very interesting. However, I am concerned that the weight license is CC BY-NC-SA 4.0. If this applies not only to the model but also to the generated images, the use cases for the model will be limited.
> 
> It seems that a pull request for optimization is being developed in the Sana repository, so I will look forward to that first.

@kohya-ss 
At least this license does not conflict with this project.
Onetrainer already supports it, why don't we?

**Comment by Bocchi-Chan2023** [comment]
I think it is a low priority because we have a better licensing model. (i.e. stable diffusion xl or flux)

**Comment by rockerBOO** [comment]
Sana updated their license to use Apache 2 on 1/11 :)

**Comment by Bocchi-Chan2023** [comment]
code only. weights are still remains very strict.

**Comment by rockerBOO** [comment]
Oh I see, the DC AE was also licensed as Apache 2 so I thought it was all going to change the license.
